independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -437,8 +442,18 @@ static void resolveInjectionPoint(BeanDeployment deployment, InjectionTargetInfo
     }
 
     static BeanInfo resolveAmbiguity(List<BeanInfo> resolved) {
-        BeanInfo selected = null;
         List<BeanInfo> resolvedAmbiguity = new ArrayList<>(resolved);
+        for (Iterator<BeanInfo> iterator = resolvedAmbiguity.iterator(); iterator.hasNext();) {
```

Comment:
Hm, this would only work for build time injection. But we will probably need something for programmatic lookup (aka `CDI.current().select()`) at runtime as well.

Reply:
Good catch! Any pointers on what part of the code I would need to update?

Reply:
I am assuming that `InstanceImp#get` needs to be updated

Reply:
Fixed